File: cslacs/Visual Studio Templates/Item Templates/vb/Templates Source/ReadOnlyRoot/ReadOnlyRoot.vb
Imports System
Imports Csla
Imports Csla.Security

namespace $rootnamespace$

  <Serializable()> _
    Public Class $safeitemname$
    Inherits ReadOnlyBase(Of $safeitemname$)
#Region "Business Methods"

    ' TODO: add your own fields, properties and methods 

    ' example with managed backing field 
    Private Shared IdProperty As PropertyInfo(Of Integer) = RegisterProperty(New PropertyInfo(Of Integer)("Id", "Id"))
    Public ReadOnly Property Id() As Integer
      Get
        Return GetProperty(IdProperty)
      End Get
    End Property

    ' example with private backing field 
    Private Shared NameProperty As PropertyInfo(Of String) = RegisterProperty(New PropertyInfo(Of String)("Name", "Name"))
    Private _name As String = NameProperty.DefaultValue
    Public ReadOnly Property Name() As String
      Get
        Return GetProperty(NameProperty, _name)
      End Get
    End Property

#End Region

#Region "Authorization Rules"

    Protected Overloads Overrides Sub AddAuthorizationRules()
      ' TODO: add authorization rules 
      'AuthorizationRules.AllowRead(NameProperty, "Role")
    End Sub

    Private Shared Sub AddObjectAuthorizationRules()
      ' TODO: add authorization rules 
      'AuthorizationRules.AllowGet(GetType($safeitemname$), "Role")
    End Sub

#End Region

#Region "Factory Methods"

    Public Shared Function Get$safeitemname$(ByVal id As Integer) As $safeitemname$
      Return DataPortal.Fetch(Of $safeitemname$)(New SingleCriteria(Of $safeitemname$, Integer)(id))
    End Function

    ' require use of factory methods 
    Private Sub New()
    End Sub

#End Region

#Region "Data Access"

    Private Overloads Sub DataPortal_Fetch(ByVal criteria As SingleCriteria(Of $safeitemname$, Integer))
      ' TODO: load values 
    End Sub

#End Region
  End Class

End Namespace

File: cslacs/Visual Studio Templates/Item Templates/cs/Templates Source/ReadOnlyRoot/ReadOnlyRoot.cs
using System;
using Csla;

namespace $rootnamespace$
{
  [Serializable]
  public class $safeitemname$ : ReadOnlyBase<$safeitemname$>
  {
    #region Business Methods

    // TODO: add your own fields, properties and methods

    // example with managed backing field
    private static PropertyInfo<int> IdProperty = 
      RegisterProperty(typeof($safeitemname$), new PropertyInfo<int>("Id", "Id"));
    public int Id
    {
      get { return GetProperty(IdProperty); }
    }

    // example with private backing field
    private static PropertyInfo<string> NameProperty = 
      RegisterProperty(typeof($safeitemname$), new PropertyInfo<string>("Name", "Name"));
    private string _name = NameProperty.DefaultValue;
    public string Name
    {
      get { return GetProperty(NameProperty, _name); }
    }

    #endregion

    #region Authorization Rules

    protected override void AddAuthorizationRules()
    {
      // TODO: add authorization rules
      //AuthorizationRules.AllowRead("Name", "Role");
    }

    private static void AddObjectAuthorizationRules()
    {
      // TODO: add authorization rules
      //AuthorizationRules.AllowGet(typeof($safeitemname$), "Role");
    }

    #endregion

    #region Factory Methods

    public static $safeitemname$ GetReadOnlyRoot(int id)
    {
      return DataPortal.Fetch<$safeitemname$>(
        new SingleCriteria<$safeitemname$, int>(id));
    }

    private $safeitemname$()
    { /* require use of factory methods */ }

    #endregion

    #region Data Access

    private void DataPortal_Fetch(SingleCriteria<$safeitemname$, int> criteria)
    {
      // TODO: load values
    }

    #endregion
  }
}
